components/blitz/src/loci/ome/io/OmeroReader.java

```diff
@@ -436,60 +459,48 @@ else if (key.equals("iid")) {
-        }
-        catch (ServerError e) {
+        catch (CannotCreateSessionException|PermissionDeniedException|ServerError e) {
             throw new FormatException(e);
         }
     }
 
     /** A simple command line tool for downloading images from OMERO. */
     public static void main(String[] args) throws Exception {
-        // parse OMERO credentials
-        BufferedReader con = new BufferedReader(
+        try (BufferedReader con = new BufferedReader(
                 new InputStreamReader(System.in, Constants.ENCODING));
-
-        System.out.print("Server? ");
-        final String server = con.readLine();
-
-        System.out.printf("Port [%d]? ", DEFAULT_PORT);
-        final String portString = con.readLine();
-        final int port = portString.equals("") ? DEFAULT_PORT :
-            Integer.parseInt(portString);
-
-        System.out.print("Username? ");
-        final String user = con.readLine();
-
-        System.out.print("Password? ");
-        final String pass = new String(con.readLine());
-
-        System.out.print("Group? ");
-        final String group = con.readLine();
-
-        System.out.print("Image ID? ");
-        final int imageId = Integer.parseInt(con.readLine());
-        System.out.print("\n\n");
-
-        // construct the OMERO reader
-        final OmeroReader omeroReader = new OmeroReader();
-        omeroReader.setUsername(user);
-        omeroReader.setPassword(pass);
-        omeroReader.setServer(server);
-        omeroReader.setPort(port);
-        omeroReader.setGroupName(group);
-        final String id = "omero:iid=" + imageId;
-        try {
+            OmeroReader omeroReader = new OmeroReader()) {
+            // parse OMERO credentials
+            System.out.print("Server? ");
+            final String server = con.readLine();
+
+            System.out.printf("Port [%d]? ", DEFAULT_PORT);
+            final String portString = con.readLine();
+            final int port = portString.equals("") ? DEFAULT_PORT :
+                Integer.parseInt(portString);
+
+            System.out.print("Username? ");
+            final String user = con.readLine();
+
+            System.out.print("Password? ");
+            final String pass = new String(con.readLine());
+
+            System.out.print("Group? ");
+            final String group = con.readLine();
+
+            System.out.print("Image ID? ");
+            final int imageId = Integer.parseInt(con.readLine());
+            System.out.print("\n\n");
+            // construct the OMERO reader
+            final String id = "omero:iid=" + imageId;
+            omeroReader.setUsername(user);
+            omeroReader.setPassword(pass);
+            omeroReader.setServer(server);
+            omeroReader.setPort(port);
+            omeroReader.setGroupName(group);
             omeroReader.setId(id);
-        }
-        catch (Exception e) {
-            omeroReader.close();
+        } catch (Exception e) {
             throw e;
```

Comment:
This should be taken care of as part of the "try-with-resources".
Can certainly double-check